bridge/engine/EngineMixer.cpp

```diff
@@ -1839,14 +1886,40 @@ void EngineMixer::onRtcpPacketDecoded(transport::RtcTransport* sender,
 SsrcOutboundContext* EngineMixer::obtainOutboundSsrcContext(size_t endpointIdHash,
     concurrency::MpmcHashmap32<uint32_t, SsrcOutboundContext>& ssrcOutboundContexts,
     const uint32_t ssrc,
-    const RtpMap& rtpMap)
+    const RtpMap& rtpMap,
+    const bool ssrcRewrite)
 {
     auto ssrcOutboundContextItr = ssrcOutboundContexts.find(ssrc);
     if (ssrcOutboundContextItr != ssrcOutboundContexts.cend())
     {
+        if (ssrcOutboundContextItr->second.markedForDeletion)
+        {
+            // The outboundContext is marked for deletion after owner inbound stream
+            // as been removed (only for non ssrcRewrite). This means the job to
+            // remove OutboundSsrc has already been enqueued and it will be deleted
+            // soon. Then we MUST NOT enqueue any more jobs with a reference to this
+            // SsrcOutboundContext
+            assert(!ssrcRewrite);
+            return nullptr;
```

Comment:
If it's marked for deletion is because `FinalizeNonSsrcRewriteOutboundContextJob` has been already enqueue. No more jobs can use this context. After `FinalizeNonSsrcRewriteOutboundContextJob` has run successfully, it will post an execution on EngineThread to delete the SsrcOuboundContext